************* Module imaplib
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1:0: C0302: Too many lines in module (1623/1000) (too-many-lines)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:25:0: C0410: Multiple imports on one line (binascii, errno, random, re, socket, subprocess, sys, time, calendar) (multiple-imports)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:41:0: C0103: Constant name "Debug" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:129:0: C0103: Constant name "_Literal" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:130:0: C0103: Constant name "_Untagged_status" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:212:8: C0103: Attribute name "Literal" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:213:8: C0103: Attribute name "Untagged_status" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:259:12: C0103: Attribute name "PROTOCOL_VERSION" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1180:8: C0103: Attribute name "mo" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:134:0: R0902: Too many instance attributes (28/7) (too-many-instance-attributes)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:180:4: C0103: Class name "error" doesn't conform to PascalCase naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:180:28: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:181:4: C0103: Class name "abort" doesn't conform to PascalCase naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:181:24: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:182:4: C0103: Class name "readonly" doesn't conform to PascalCase naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:182:27: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:184:23: W0621: Redefining name 'host' from outer scope (line 1542) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:274:0: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:291:8: W0621: Redefining name 'host' from outer scope (line 1542) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:295:19: W0621: Redefining name 'host' from outer scope (line 1542) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:364:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:433:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:445:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:466:17: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:524:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:539:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:548:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:556:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:567:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:576:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:590:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:601:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:617:4: C0103: Method name "_CRAM_MD5_AUTH" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:619:8: C0415: Import outside toplevel (hmac) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:633:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:646:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:654:12: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:663:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:686:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:699:8: W0612: Unused variable 'name' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:723:17: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:747:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:769:0: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:773:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:782:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:796:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:811:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:810:26: W0212: Access to a protected member _create_stdlib_context of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:831:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:842:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:860:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:864:0: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:881:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:897:0: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:919:36: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:922:8: C0103: Variable name "ur" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:939:0: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:960:28: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:939:4: R0912: Too many branches (23/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1035:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1035:8: W0612: Unused variable 'typ' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1061:12: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1084:28: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1085:21: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1043:4: R0912: Too many branches (15/12) (too-many-branches)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1149:12: W0612: Unused variable 'val' (unused-variable)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1175:4: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1195:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1203:0: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1208:38: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1222:8: C0103: Argument name "s" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1225:12: C0103: Variable name "tm" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1222:8: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1229:27: W0622: Redefining built-in 'dict' (redefined-builtin)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1231:12: C0103: Variable name "l" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1232:22: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1233:12: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1234:12: C0103: Variable name "l" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1246:15: C0103: Variable name "n" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1250:16: W0702: No exception type(s) specified (bare-except)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1255:16: C0103: Variable name "n" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:217:8: W0201: Attribute 'utf8_enabled' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:218:8: W0201: Attribute '_encoding' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:219:8: W0201: Attribute 'Literal' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:220:8: W0201: Attribute 'Untagged_status' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1242:16: W0201: Attribute '_cmd_log_idx' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:813:12: W0201: Attribute 'sock' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:815:12: W0201: Attribute 'file' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:613:8: W0201: Attribute 'user' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:613:19: W0201: Attribute 'password' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1040:8: W0201: Attribute 'capabilities' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1180:8: W0201: Attribute 'mo' defined outside __init__ (attribute-defined-outside-init)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:134:0: R0904: Too many public methods (50/20) (too-many-public-methods)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1260:4: C0103: Class name "IMAP4_SSL" doesn't conform to PascalCase naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1279:27: W0621: Redefining name 'host' from outer scope (line 1542) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1279:8: R0913: Too many arguments (6/5) (too-many-arguments)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1288:16: C0415: Import outside toplevel (warnings) (import-outside-toplevel)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1304:23: W0621: Redefining name 'host' from outer scope (line 1542) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1315:0: C0103: Class name "IMAP4_stream" doesn't conform to PascalCase naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1315:0: R0902: Too many instance attributes (8/7) (too-many-instance-attributes)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1332:19: W0621: Redefining name 'host' from outer scope (line 1542) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1401:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1404:16: C0103: Variable name "t" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1406:12: C0103: Variable name "e" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1387:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1411:4: R0201: Method could be a function (no-self-use)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1419:0: C0103: Function name "Internaldate2tuple" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1426:4: W0621: Redefining name 'mo' from outer scope (line 1596) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1436:4: W0622: Redefining built-in 'min' (redefined-builtin)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1426:4: C0103: Variable name "mo" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1447:4: C0103: Variable name "tt" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1454:0: C0103: Function name "Int2AP" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1458:4: W0621: Redefining name 'val' from outer scope (line 1529) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1458:15: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1458:15: C0103: Variable name "AP" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1467:0: C0103: Function name "ParseFlags" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1471:4: W0621: Redefining name 'mo' from outer scope (line 1596) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1471:4: C0103: Variable name "mo" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1478:0: C0103: Function name "Time2Internaldate" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1491:8: C0103: Variable name "dt" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1505:8: C0103: Variable name "dt" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1509:8: C0103: Variable name "dt" doesn't conform to snake_case naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1525:4: C0410: Multiple imports on one line (getopt, getpass) (multiple-imports)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1532:4: C0103: Constant name "stream_command" doesn't conform to UPPER_CASE naming style (invalid-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1538:25: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1540:17: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1575:12: W0621: Redefining name 'cmd' from outer scope (line 1592) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1575:17: W0621: Redefining name 'args' from outer scope (line 1528) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1577:13: W0621: Redefining name 'dat' from outer scope (line 1602) (redefined-outer-name)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1576:8: W0212: Access to a protected member _mesg of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1578:8: W0212: Access to a protected member _mesg of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1579:24: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1589:8: W0212: Access to a protected member _mesg of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1590:8: W0212: Access to a protected member _mesg of a client class (protected-access)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1597:19: C0321: More than one statement on a single line (multiple-statements)
/log/firmware/firmware/usr/lib/python3.8/imaplib.py:1608:24: C0321: More than one statement on a single line (multiple-statements)

-----------------------------------
Your code has been rated at 8.21/10

